ggml.h

```diff
@@ -364,11 +364,12 @@ extern "C" {
         GGML_TYPE_IQ3_S   = 21,
         GGML_TYPE_IQ2_S   = 22,
         GGML_TYPE_IQ4_XS  = 23,
-        GGML_TYPE_I8      = 24,
-        GGML_TYPE_I16     = 25,
-        GGML_TYPE_I32     = 26,
-        GGML_TYPE_I64     = 27,
-        GGML_TYPE_F64     = 28,
+        GGML_TYPE_IQ1_M   = 24,
+        GGML_TYPE_I8      = 25,
+        GGML_TYPE_I16     = 26,
+        GGML_TYPE_I32     = 27,
+        GGML_TYPE_I64     = 28,
+        GGML_TYPE_F64     = 29,
```

Comment:
Need to also update the enum in `gguf-py/gguf/constants.py`:

https://github.com/ggerganov/llama.cpp/blob/deb7240100da99555b9ab9dc635021e591fceaf5/gguf-py/gguf/constants.py#L681-L708

Also, move `GGML_TYPE_IQ1_M` at the end of the enum to keep backwards compatibility with any GGUF files that might have started using integer of 64-bit types